File: 1291-sequential-digits/1291-sequential-digits.cpp
class Solution {
public:
    vector<int> sequentialDigits(int low, int high) {
        string digits = "123456789";
        vector<int>res;

        for(int length = 2 ; length <= 9 ; length++){
            for(int start = 0; start <= 9 - length ; start ++){
                string sub = digits.substr(start , length);
                int num = stoi(sub);

                if(num >= low && num <= high){
                    res.push_back(num);
                }
                else if (num > high){
                    break;
                }
            }
        }

        return res;
    }
};

File: 1291-sequential-digits/1291-sequential-digits.py
class Solution:
    def sequentialDigits(self, low: int, high: int) -> List[int]:
        res = []
        digits = '123456789'

        for length in range(2 , 10):
            for start in range( 10 - length):
                num = int(digits[start : start + length])

                if low <= num <= high:
                    res.append(num)
                elif num > high:
                    break
        
        return res